pkg/api/v1/pod/util.go

```diff
@@ -432,16 +432,16 @@ func GetPodObservedGenerationIfEnabled(pod *v1.Pod) int64 {
 // We will emit condition.observedGeneration if the feature is enabled OR if condition.observedGeneration is already set.
 // This protects against an infinite loop of kubelet trying to clear the value after the FG is turned off, and
 // the API server preserving existing values when an incoming update tries to clear it.
-func GetPodObservedGenerationIfEnabledOnCondition(pod *v1.Pod, conditionType v1.PodConditionType) int64 {
-	if pod == nil {
+func GetPodObservedGenerationIfEnabledOnCondition(podStatus *v1.PodStatus, generation int64, conditionType v1.PodConditionType) int64 {
```

Comment:
(preexisting / non-blocking)

rereading this, the function name / godoc on this function and the function above are confusing... it *sounds* like they will return observedGeneration, but they are returning generation

I'm having a hard time coming up with an alternative name that isn't ridiculously long, but I think the current names will lead to callers accidentally using these incorrectly

as a follow-up, consider if we can name these better to make it clearer the thing being returned is the passed-in generation or 0

Reply:
`CalculatePodStatusObservedGeneration` and `CalculatePodConditionObservedGeneration` maybe?